src/SFA.DAS.Payments.MatchedLearner.Application/Repositories/PaymentsDataLockRepository.cs

```diff
@@ -29,70 +30,72 @@ public async Task<MatchedLearnerDataLockInfo> GetDataLockEvents(long ukprn, long
-                .Where(y => y.Ukprn == ukprn)
-                .OrderByDescending(y => y.AcademicYear)
-                .ThenByDescending(y => y.CollectionPeriod)
-                .FirstOrDefaultAsync();
-
-            if (latestSuccessfulJob == null)
+            var latestSuccessfulJobs = await _context.LatestSuccessfulJobs
+                .Where(x => x.Ukprn == ukprn)
+                .Select(l => l.DcJobId)
+                .Distinct()
+                .ToListAsync();
+            
+            if (!latestSuccessfulJobs.Any())
             {
                 stopwatch.Stop();
-                _logger.LogInformation($"No Data in current academic year CollectionPeriod for Uln: {uln}, Duration: {stopwatch.ElapsedMilliseconds}");
+                _logger.LogInformation($"No Data for Uln: {uln}, Duration: {stopwatch.ElapsedMilliseconds}");
                 return new MatchedLearnerDataLockInfo();
             }
 
-            _logger.LogDebug($"Getting DataLock Event Data Uln: {uln}, Academic year: {latestSuccessfulJob.AcademicYear}, Collection period: {latestSuccessfulJob.CollectionPeriod}");
-
+            _logger.LogDebug($"Getting DataLock Event Data for Uln: {uln}");
+          
             var transactionTypes = new List<byte> { 1, 2, 3 };
 
             var dataLockEvents = await _context.DataLockEvent
                 .Where(x =>
                     x.LearningAimReference == "ZPROG001" &&
                     x.Ukprn == ukprn &&
                     x.LearnerUln == uln &&
-                    x.JobId == latestSuccessfulJob.DcJobId &&
-                    x.AcademicYear == latestSuccessfulJob.AcademicYear &&
-                    x.CollectionPeriod == latestSuccessfulJob.CollectionPeriod)
+                    latestSuccessfulJobs.Contains(x.JobId))
                 .OrderBy(x => x.LearningStartDate)
                 .ToListAsync();
 
-            var eventId = dataLockEvents.Select(d => d.EventId).ToList();
+            var eventIds = dataLockEvents.Select(d => d.EventId).ToList();
 
             var dataLockEventPriceEpisodes = await _context.DataLockEventPriceEpisode
-                .Where(d => eventId.Contains(d.DataLockEventId) && d.PriceEpisodeIdentifier != null)
+                .Where(d => eventIds.Contains(d.DataLockEventId) && d.PriceEpisodeIdentifier != null)
                 .OrderBy(p => p.StartDate)
+                .ThenBy(p => p.PriceEpisodeIdentifier)
                 .ToListAsync();
 
-
             var dataLockEventPayablePeriods = await _context.DataLockEventPayablePeriod
-                .Where(d => eventId.Contains(d.DataLockEventId) && transactionTypes.Contains(d.TransactionType) && d.PriceEpisodeIdentifier != null && d.Amount != 0)
+                .Where(d => eventIds.Contains(d.DataLockEventId) && transactionTypes.Contains(d.TransactionType) && d.PriceEpisodeIdentifier != null && d.Amount != 0)
                 .OrderBy(p => p.DeliveryPeriod)
                 .ToListAsync();
 
```

Comment:
As this table is quite large probably worth checking this is covered by an index too.